mount/mount_windows.go

```diff
@@ -101,5 +163,30 @@ func Unmount(mount string, flags int) error {
 
 // UnmountAll unmounts from the provided path
 func UnmountAll(mount string, flags int) error {
+	if mount == "" {
+		// This isn't an error, per the EINVAL handling in the Linux version
+		return nil
+	}
+
 	return Unmount(mount, flags)
 }
+
+func (m *Mount) bindMount(target string) error {
+	for _, option := range m.Options {
+		if option == "ro" {
+			return errors.Wrap(ErrNotImplementOnWindows, "read-only bind mount")
+		}
+	}
+
+	if err := os.Remove(target); err != nil {
+		return err
+	}
+
+	// TODO: We don't honour the Read-Only flag.
+	// It's possible that Windows simply lacks this.
```

Comment:
TODO: Remove this comment, and fix the description... We actively reject read-only bind mounts earlier in this function, because we will no longer generate them in the Snapshot code: An attempt to mount a committed layer read-only will instead create a small temporary child layer and mount _that_, so it won't be a bind-mount.

In fact, the only time we'll see a bind-mount from the Windows snapshotter is for a read-write parentless layer, i.e. `FROM scratch`-type usage for arbitrary data transfer.

The comments and commit description are legacy from an earlier version were the snapshotter would bind-mount any read-only layer that was requested, hoping the user wouldn't then edit the data and corrupt it. It was not a good approach...